respx/models.py

```diff
@@ -383,13 +374,13 @@ def match(self, request: Request) -> Optional[Union[Request, ResponseTemplate]]:
         if self.method != request_method.decode():
             return None
 
-        request_url = self.build_url(_request_url)
-        if not self._url:
+        if not self.url:
             matches = True
-        elif isinstance(self._url, str):
-            matches = self._url == request_url
+        elif isinstance(self.url, httpx.URL):
+            matches = self.url.raw == _request_url
         else:
-            match = self._url.match(request_url)
+            request_url = build_url(_request_url)
+            match = self.url.match(str(request_url))
```

Comment:
`_request_url` will always be a raw tuple, i.e. no need for calling `build_url`, right?

Would this be enough?
```suggestion
            match = self.url.match(str(httpx.URL(_request_url)))
```

If so, var could be named `request_url` above.